src/main/java/org/testng/TestRunner.java

```diff
@@ -99,14 +100,6 @@
       Maps.newHashMap();
   private final DataProviderHolder holder = new DataProviderHolder();
 
-  /**
-   * All the test methods we found, associated with their respective classes. Note that these test
-   * methods might belong to different classes. We pick which ones to run at runtime.
-   */
-  private ITestNGMethod[] m_allTestMethods = new ITestNGMethod[0];
```

Comment:
I propose to keep the attribute as a cache for the moment and only skip it when `RuntimeBehavior.isMemoryFriendlyMode()` is used.
It will allow us to have feedbacks without taking the risk of regressions.

Reply:
yes, but in either case, this variable is redundant and is now going to be relevant only when the user runs in JUnit mode.
In all other times, we will now query `m_groupMethods` to get the test methods because we instantiate this object by computing `m_allTestMethods` and so, we don't need `m_allTestMethods` anymore (All it does is cause two references to the same set of memory allocations).

So now all the logic of what to query, and return back resides within `getAllTestMethods()` method.